### ISSUE TITLE
Dev healthcheck failing - nslookup: command not found DNS resolution failed

### ISSUE THREAD

**Issue by ilike2burnthing** [haugene/docker-transmission-openvpn#1796]
## Before creating this issue I have:

**REQUIRED**
- [x] Read through the pinned issues for related problems
- [x] Searched for similar [provider issues][pi] and [container issues][ci]
- [x] [Read the documentation][rd], especially the troubleshooting section and FAQ
- [x] Tried to add as much relevant information to the issue as possible
- [x] Verified I have tried using newest release as well
- [x] Agreed that my issue will be closed if I do not follow this template and will remain closed until I complete the template

[pi]: https://github.com/haugene/vpn-configs-contrib/issues
[ci]: https://github.com/haugene/docker-transmission-openvpn/issues
[rd]: https://haugene.github.io/docker-transmission-openvpn/

### Container version & last working release

**Required, problem occurs in :** 
 
```dev (old image deleted, fresh image pulled ~30mins ago)```

*If possible, last working version:* 
 
```latest (old image deleted, fresh image pulled ~30mins ago), and dev a day or two ago (I use Watchtower for updates)```

### Describe the problem

**REQUIRED**
```txt
Healthcheck repeatedly fails and container state remains as [starting].

Healthcheck output - /etc/scripts/healthcheck.sh: line 15: nslookup: command not found DNS resolution failed

Transmission itself seems to be working fine, able to add a new magnet link and begin downloading. Magnet from https://ipleak.net/ shows VPN IP.
```
### Describe the steps you have tried to solve the problem

**REQUIRED**
```txt
Changed image to [latest], works, changed back to dev (fresh image), same problem.

Couldn't find any similar issues, but given that changing the branch resolves it, I'm guessing it's a result of recent changes.

Running [nslookup google.com] in the container results in [bash: nslookup: command not found]
```

### Add your docker run command or docker-compose file or env details

**REQUIRED**
```txt
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "dumb-init",
            "/etc/openvpn/start.sh"
        ],
        "Domainname": "",
        "Entrypoint": null,
        "Env": [
            "PUID=999",
            "PGID=999",
            "OPENVPN_PROVIDER=NORDVPN",
            "OPENVPN_USERNAME=removed",
            "OPENVPN_PASSWORD=removed",
            "LOCAL_NETWORK=192.168.0.0/24",
            "TRANSMISSION_DOWNLOAD_DIR=/media/Complete",
            "TRANSMISSION_INCOMPLETE_DIR=/media/Incomplete",
            "LOG_TO_STDOUT=true",
            "NORDVPN_COUNTRY=IE",
            "NORDVPN_CATEGORY=P2P",
            "NORDVPN_PROTOCOL=UDP",
            "OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60 --txqueuelen 10000 --mssfix 1472",
            "TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true",
            "TRANSMISSION_RPC_PASSWORD=removed",
            "TRANSMISSION_RPC_USERNAME=removed",
            "TRANSMISSION_TRASH_ORIGINAL_TORRENT_FILES=true",
            "TRANSMISSION_WEB_UI=flood-for-transmission",
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "GLOBAL_APPLY_PERMISSIONS=true",
            "TRANSMISSION_HOME=/data/transmission-home",
            "TRANSMISSION_RPC_PORT=9091",
            "TRANSMISSION_WATCH_DIR=/data/watch",
            "CREATE_TUN_DEVICE=true",
            "ENABLE_UFW=false",
            "UFW_ALLOW_GW_NET=false",
            "UFW_DISABLE_IPTABLES_REJECT=false",
            "WEBPROXY_ENABLED=false",
            "WEBPROXY_PORT=8118",
            "HEALTH_CHECK_HOST=google.com",
            "UFW_EXTRA_PORTS=",
            "DROP_DEFAULT_ROUTE=",
            "WEBPROXY_USERNAME=",
            "WEBPROXY_PASSWORD=",
            "REVISION=0667bbddc1381b4ac9341009a449fe6b8186e1b4"
        ],
        "ExposedPorts": {
            "51413/tcp": {},
            "51413/udp": {},
            "8118/tcp": {},
            "9091/tcp": {}
        },
        "Healthcheck": {
            "Interval": 60000000000,
            "Test": [
                "CMD-SHELL",
                "/etc/scripts/healthcheck.sh"
            ]
        },
        "Hostname": "d076bc14c82f",
        "Image": "haugene/transmission-openvpn:dev",
        "Labels": {
            "autoheal": "true",
            "org.opencontainers.image.revision": "0667bbddc1381b4ac9341009a449fe6b8186e1b4",
            "org.opencontainers.image.source": "https://github.com/haugene/docker-transmission-openvpn"
        },
        "OnBuild": null,
        "OpenStdin": true,
        "StdinOnce": false,
        "Tty": true,
        "User": "",
        "Volumes": {
            "/config": {},
            "/data": {},
            "/etc/localtime": {},
            "/media": {}
        },
        "WorkingDir": ""
```

### Logs

 

**REQUIRED**
```txt
Transmission startup script complete.
Wed Mar 31 17:22:03 2021 /sbin/ip route add xx.xx.xx.xx/32 via 172.17.0.1
Wed Mar 31 17:22:03 2021 /sbin/ip route add 0.0.0.0/1 via xx.xx.xx.xx
Wed Mar 31 17:22:03 2021 /sbin/ip route add 128.0.0.0/1 via xx.xx.xx.xx
Wed Mar 31 17:22:03 2021 Initialization Sequence Completed
[2021-03-31 17:22:04.891] Transmission 2.94 (d8e60ee44f) started (session.c:740)
[2021-03-31 17:22:04.892] RPC Server Adding address to whitelist: 127.0.0.1 (rpc-server.c:971)
[2021-03-31 17:22:04.892] RPC Server Adding address to whitelist: ::1 (rpc-server.c:971)
[2021-03-31 17:22:04.892] RPC Server Serving RPC and Web requests on port 127.0.0.1:9091/transmission/ (rpc-server.c:1213)
[2021-03-31 17:22:04.892] RPC Server Password required (rpc-server.c:1220)
[2021-03-31 17:22:04.892] Port Forwarding Stopped (port-forwarding.c:180)
[2021-03-31 17:22:04.892] DHT Reusing old id (tr-dht.c:307)
[2021-03-31 17:22:04.892] DHT Bootstrapping from 41 IPv4 nodes (tr-dht.c:156)
[2021-03-31 17:22:04.892] Using settings from "/data/transmission-home" (daemon.c:528)
[2021-03-31 17:22:04.892] Saved "/data/transmission-home/settings.json" (variant.c:1266)
[2021-03-31 17:22:04.892] transmission-daemon requiring authentication (daemon.c:554)
[2021-03-31 17:22:04.892] Watching "/data/watch" for new .torrent files (daemon.c:573)
```

### Host system

**REQUIRED**
```txt
NAS, Docker CE 19.03.8
```

**Comment by coreyp93** [comment]
Im having the same issue, on docker too latest dev. Im worried, does this mean if healthcheck is not working the container could continue to work even if not connected to the VPN meaning the traffic gets exposed? Making this a rather serious bug. It only seems to have happened fairly recently so has there been some breaking change? Just dont want to be running an exposed container. Thanks.

**Comment by ilike2burnthing** [comment]
The container is still connected to the VPN. You can check using the magnet link here - https://ipleak.net/

**Comment by coreyp93** [comment]
Yes I did already check and I am connected to the VPN right now, but I was asking should the VPN go down the expected behaviour is that the container should stop working to prevent exposing data, and was worried while the health check is failing should the VPN go down the container continue working. I was just checking if that is what's responsible for the containers failsafe or not. Usually it pings for disconnects and I'm assuming right now it isn't doing that, therefore unable to kill the connection should the VPN go down for whatever reason.

**Comment by ilike2burnthing** [comment]
Oh, sorry, get you now. Yea, that's cause for concern.

**Comment by ilike2burnthing** [comment]
@haugene @pkishino guessing you're both busy at the moment, but just so you're aware.

**Comment by pkishino** [comment]
@haugene probably the change to ubuntu base is missing some dependencies..
quite busy atm , perhaps next week I can take a look

**Comment by kyleinprogress** [comment]
Not sure why, but the alpine image no longer has nslookup in it. It needs to be added to the container during the build process by adding **bind-tools** via APK.

Github won't let me create a pull request because it's trying to add all the github actions in my fork which I don't want to mess with. Should be a relatively simple change.

https://github.com/haugene/docker-transmission-openvpn/blob/cf529b9b776589279f4aeb325a0b05b1745e2ac1/Dockerfile#L6-L8

```
 RUN echo "@community http://dl-cdn.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories \ 
     && apk --no-cache add bash dumb-init ip6tables ufw@community openvpn shadow transmission-daemon transmission-cli \ 
         curl jq tzdata openrc tinyproxy tinyproxy-openrc openssh unrar git bind-tools \
```

**Comment by pkishino** [comment]
Not sure why you can’t make a PR, no problems here .
We’re moving away from alpine with the next release though so let’s see

**Comment by mustanggb** [comment]
Created pull request #1806.

**Comment by pkishino** [comment]
Fixed in #1806  on dev branch
